game/src/AI/States/Charge.gd

```diff
@@ -0,0 +1,37 @@
+extends State
+"""
+State machine state to control the charge attack.
+Controls the amount of acceleration and speed from the straight line behavior's
+desired velocity, and only charges for a set amount of time before moving to cooldown.
+"""
+
+onready var _behavior: = get_node(behavior) as StraightLineBehavior2D
+
+export var behavior: = NodePath()
+export var charge_time: = 1.0
+export var time_to_full_speed = 0.5
+
+var _steering: = SteeringMotion2D.new()
+
+func enter(msg: Dictionary = {}) -> void:
+	var target_shape: = msg.get("target") as CollisionShape2D
+	_behavior.target = (target_shape.global_position - _behavior.controller.actor.global_position).normalized() * 10000
+	
+	$Timer.connect("timeout", self, "_on_Timer_timeout")
```

Comment:
We always put node dependencies in `onready var` at the top of the script:

1. This way all dependencies are visible as soon as you open the script
2. get_node gets heavy in a growing project (not here but if you get used to doing that in process callbacks), better cache it by default